icu4j/main/classes/core/src/com/ibm/icu/number/NumberFormatterImpl.java

```diff
@@ -244,6 +236,18 @@ private static MicroPropsGenerator macrosToMicroGenerator(MacroProps macros, Mic
         /// START POPULATING THE DEFAULT MICROPROPS AND BUILDING THE MICROPROPS GENERATOR ///
         /////////////////////////////////////////////////////////////////////////////////////
 
+        // Unit Preferences and Conversions as our first step
+        if (macros.usage != null) {
+            if (!isCldrUnit) {
+                throw new AssertionError("We only support \"usage\" when the input unit is specified, and is a CLDR Unit.");
```

Comment:
This is a user error, so IllegalArgumentException, not AssertionError

Reply:
done